Kernel/Arch/i386/CPU.cpp

```diff
@@ -297,6 +297,26 @@ void page_fault_handler(RegisterState regs)
     }
 }
 
+EH_ENTRY_NO_CODE(1, debug);
+void debug_handler(RegisterState regs)
+{
+    clac();
+    if (!Process::current || Process::current->is_ring0()) {
```

Comment:
`Process::is_ring0` here is a bit deceptive.
It only tells us whether the current process is a "ring 0 process" AKA a kernel process.
It doesn't look at the current value of the CS register.
As such, this is not enough protection against trapping in ring 0. We need to check `regs.cs` :)